src/test/rustdoc/intra-doc/non-path-primitives.rs

```diff
@@ -8,6 +8,15 @@
 // @has - '//a[@href="https://doc.rust-lang.org/nightly/std/primitive.array.html#method.map"]' 'array::map'
 //! [array::map]
 
+// @has - '//a[@href="https://doc.rust-lang.org/nightly/std/primitive.str.html"]' 'owned str'
+// @has - '//a[@href="https://doc.rust-lang.org/nightly/std/primitive.str.html"]' 'str ref'
+// @has - '//a[@href="https://doc.rust-lang.org/nightly/std/primitive.str.html#method.is_empty"]' 'str::is_empty'
+// @has - '//a[@href="https://doc.rust-lang.org/nightly/std/primitive.str.html#method.len"]' '&str::len'
+//! [owned str][str]
+//! [str ref][&str]
+//! [str::is_empty]
+//! [&str::len]
```

Comment:
The one thing that gives me pause is that `&str::len` is a valid link. However, things like `<&str>::default()` are valid Rust code, so it's not too different. Also, we treat `*::is_null` as a valid link, which is not even close to valid Rust code, so it's not like there isn't precedent
.

Reply:
>  Also, we treat *::is_null as a valid link, which is not even close to valid Rust code,

I think we should remove that actually.

Reply:
Yeah, it seems confusing.

Reply:
> > Also, we treat *::is_null as a valid link, which is not even close to valid Rust code,
> 
> I think we should remove that actually.

Opened #80892.